tests/commonTemplates_test.go

```diff
@@ -354,6 +328,125 @@ var _ = Describe("Common templates", func() {
 	})
 })
 
+var _ = Describe("Common templates with multiple architectures enabled", Ordered, func() {
+	var (
+		testTemplateAmd64 testResource
+		testTemplateArm64 testResource
+		testTemplateS390x testResource
+	)
+
+	// Using ordered container and BeforeAll for performance reasons.
+	// These tests don't need to disable and enable multi-arch before every test.
+	BeforeAll(func() {
+		strategy.SkipSspUpdateTestsIfNeeded()
+
+		const templateName = "rhel9-server-small"
+		expectedLabels := expectedLabelsFor("common-templates", common.AppComponentTemplating)
+
+		testTemplateAmd64 = testResource{
+			Name:           templateName,
+			Namespace:      strategy.GetTemplatesNamespace(),
+			Resource:       &templatev1.Template{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(t *templatev1.Template) {
+				t.Parameters = nil
+			},
+			EqualsFunc: func(old *templatev1.Template, new *templatev1.Template) bool {
+				return reflect.DeepEqual(old.Parameters, new.Parameters)
+			},
+		}
+
+		testTemplateArm64 = testResource{
+			Name:           templateName + "-" + arm64,
+			Namespace:      strategy.GetTemplatesNamespace(),
+			Resource:       &templatev1.Template{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(t *templatev1.Template) {
+				t.Parameters = nil
+			},
+			EqualsFunc: func(old *templatev1.Template, new *templatev1.Template) bool {
+				return reflect.DeepEqual(old.Parameters, new.Parameters)
+			},
+		}
+
+		testTemplateS390x = testResource{
+			Name:           templateName + "-" + s390x,
+			Namespace:      strategy.GetTemplatesNamespace(),
+			Resource:       &templatev1.Template{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(t *templatev1.Template) {
+				t.Parameters = nil
+			},
+			EqualsFunc: func(old *templatev1.Template, new *templatev1.Template) bool {
+				return reflect.DeepEqual(old.Parameters, new.Parameters)
+			},
+		}
+
+		updateSsp(func(foundSsp *ssp.SSP) {
+			foundSsp.Spec.EnableMultipleArchitectures = ptr.To(true)
+			foundSsp.Spec.Cluster = &ssp.Cluster{
+				WorkloadArchitectures:     []string{amd64, arm64, s390x},
+				ControlPlaneArchitectures: []string{nodeArchitecture},
+			}
+		})
+		waitUntilDeployed()
+	})
+
+	AfterAll(func() {
+		strategy.RevertToOriginalSspCr()
+	})
+
+	DescribeTable("created resource", func(res *testResource) {
+		Expect(apiClient.Get(ctx, res.GetKey(), res.NewResource())).To(Succeed())
```

Comment:
here we check for resource to be created is it worth to check label TemplateArchitectureLabel here